sdk/storage/azure-storage-blob/tests/test_blob_tags.py

```diff
@@ -384,6 +384,34 @@ def test_filter_blobs(self, storage_account_name, storage_account_key):
         self.assertEqual(items_on_page2[0]['tags']['tag1'], 'firsttag')
         self.assertEqual(items_on_page2[0]['tags']['tag2'], 'secondtag')
 
+    @BlobPreparer()
+    def test_filter_blobs_versions(self, versioned_storage_account_name, versioned_storage_account_key):
+        self._setup(versioned_storage_account_name, versioned_storage_account_key)
+
+        blob_name = self._get_blob_reference()
+        blob_client = self.bsc.get_blob_client(self.container_name, blob_name)
+        tags1 = {"tag1": "firsttag"}
+        include_list = ['versions']
+        where = "\"tag1\"='firsttag'"
+
+        # Act
+        blob_client.create_append_blob(tags=tags1)
+        blob_client.create_append_blob(tags=tags1)
+
+        if self.is_live:
+            sleep(10)
+
+        blob_list = self.bsc.find_blobs_by_tags(filter_expression=where, results_per_page=2,
```

Comment:
Sorry I probably should have mentioned this while we were working on it, but would you mind refactoring the tests to not use `by_page`? That's really only to test pagination which is covered by other tests so this test should be simplified and only test the version support. For sync it would be pretty simple. Just make this line more like (just wrapping the result in `list()` and not including the `results_per_page` arg):

`blob_list = list(self.bsc.find_blobs_by_tags(filter_expression=where, include=include_list))`

Then just minor refactoring of the asserts.

Reply:
Same comment applies to async and container versions of the tests.